20_classes_lab/03.py

```diff
@@ -1,13 +1,49 @@
-"""
-Following code assumes a class named Widget that depends on
-other widgets
-Widget ctor takes a name, and function 
-add_dependency(depends)
-marks widget as depends on all widgets in "depends" list
-
-When building a widget, you need to first build all the widgets it
-depends on.
-
-Implement widget so the following code works
-"""
-
+"""
+      _      _      _
+   __(.)< __(.)> __(.)=
+   \___)  \___)  \___)   
+
+Python course - Verint
+Week 3
+
+"""
+
+class Widget(object):
+
+    def __init__(self, name):
+        self.name = name
+        self.depends = []
+        self.complete = False
+
+    def add_dependency(self, *Widget):
+        self.depends.extend(Widget)
+
+    def build(self):
+        self.complete = True
+        for dep in self.depends:
+            if not dep.complete:

```

Comment:
Fantastic! One style comment: It's usually nicer to `continue` on wrong input than to perform on the right ones. So prefer:

```
for dep in self.depends:
  if dep.complete: continue
 
  dep.build()
  print dep.name
```